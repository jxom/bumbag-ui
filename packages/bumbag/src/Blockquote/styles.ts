import { cssClass } from '../styled';
import { darken, space, theme } from '../utils';

export const Blockquote = (styleProps) => cssClass`
  border-left: 4px solid ${darken(0.2, 'white')(styleProps)};
  padding: ${space(4)(styleProps)}rem;

  & {
    ${theme(styleProps.themeKey, `styles.base`)(styleProps)};
  }
`;
