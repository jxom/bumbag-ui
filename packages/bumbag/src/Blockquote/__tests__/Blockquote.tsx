import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Blockquote } from '../Blockquote';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Blockquote>Hello world</Blockquote>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Blockquote borderColor="primary">Hello world</Blockquote>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('composition', () => {
  describe('as', () => {
    it.only('should render correctly', () => {
      const { container } = render(<Blockquote use="p">Hello world</Blockquote>);
      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe('hook', () => {
    it('should return with Blockquote props', () => {
      const { result } = renderHook(() => Blockquote.useProps());
      expect(result.current).toMatchSnapshot();
    });
  });

  describe('render props', () => {
    it('should render correctly', () => {
      const { container } = render(
        <Blockquote>{(BlockquoteProps) => <div {...BlockquoteProps}>Hello world</div>}</Blockquote>
      );
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});

describe('theming', () => {
  it('Blockquote.base should render correctly', () => {
    const { container } = render(<Blockquote>hello world</Blockquote>, {
      theme: { Blockquote: { styles: { base: { backgroundColor: 'red' } } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly', () => {
    const { container } = render(<Blockquote>hello world</Blockquote>, {
      theme: { Blockquote: { defaultProps: { className: 'test' } } },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
