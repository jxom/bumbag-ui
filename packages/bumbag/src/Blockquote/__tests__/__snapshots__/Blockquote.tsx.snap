// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition as should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border-left: 4px solid #cccccc;
  padding: 1rem;
}

<p
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Blockquote"
>
  Hello world
</p>
`;

exports[`composition hook should return with Blockquote props 1`] = `
Object {
  "className": "css-hqv8z2 css-1pwrqjj css-3ekj6a fp-Box fp-Blockquote",
}
`;

exports[`composition render props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border-left: 4px solid #ccc;
  padding: 1rem;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

<div
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Blockquote"
>
  Hello world
</div>
`;

exports[`defaultProps should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border-left: 4px solid #ccc;
  padding: 1rem;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

<blockquote
  class="emotion-0 emotion-1 emotion-2 test fp-Box fp-Blockquote"
>
  hello world
</blockquote>
`;

exports[`props should render correctly 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border-left: 4px solid #ccc;
  padding: 1rem;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

<blockquote
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Blockquote"
>
  Hello world
</blockquote>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-1 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-1:focus:not(:focus-visible) {
  outline: none;
}

.emotion-0 {
  border-left: 4px solid #ccc;
  padding: 1rem;
}

.emotion-0:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-2 {
  border-color: #3722ed !important;
}

<blockquote
  class="emotion-0 emotion-1 emotion-2 fp-Box fp-Blockquote"
>
  Hello world
</blockquote>
`;
